src/error_chain.rs

```diff
@@ -137,6 +137,17 @@ macro_rules! error_chain_processed {
                 )
             }
 
+            /// Construct a chained error from another boxed error and a kind, and generates a backtrace
+            pub fn with_boxed_chain<K>(error: Box<::std::error::Error + Send>, kind: K)
+                -> $error_name
+                where K: Into<$error_kind_name>
+            {
+                $error_name(
+                    kind.into(),
+                    $crate::State::new::<$error_name>(error, ),
+                )
+            }
+
```

Comment:
I spent a long time trying to find a solution to this. The only options that I've gotten to work involve introducing a new method on `Error`, or introducing a second `ResultExt`-alike. Apparently, supporting patterning-matching on chained `Error`s as well as boxed error types is quite challenging with Rust's type system.